pycbc/waveform/waveform.py

```diff
@@ -760,27 +760,20 @@ def _base_get_td_waveform_from_fd(template=None, rwrap=0.2, **params):
     if nparams['approximant'] not in fd_det:
```

Comment:
@WuShichao Actuallym, why is there this check here and two paths? The first should work for both cases as all it does is some checks of the arguments and the calls the function in _filter_time_lengths already. Can this duplication be removed?